### PR TITLE
[Refactor]: refactor MAE visualization

### PR DESCRIPTION
**Pull request by soonera**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Refactor MAE visualization

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by soonera** [comment]
The visualization is as follows：

**Comment by fangyixiao18** [comment]
the [mae_visualization.py](https://github.com/open-mmlab/mmselfsup/blob/master/tools/misc/mae_visualization.py) hasn't been added?

File: mmselfsup/models/algorithms/mae.py [hunk 1]
```diff
@@ -30,6 +31,39 @@ def extract_feat(self, inputs: List[torch.Tensor],
         pred = self.neck(latent, ids_restore)
         return pred
 
+    def predict(self,
+                inputs: List[torch.Tensor],
+                data_samples: Optional[List[SelfSupDataSample]] = None,
+                **kwargs) -> SelfSupDataSample:
+        """The forward function in testing.
```
Comment (fangyixiao18) [inline mmselfsup/models/algorithms/mae.py]: Add docstring to indicate the function is mainly for image reconstruction

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: mmselfsup/models/algorithms/mae.py (hunk 1 above)
Comment (soonera) [reply mmselfsup/models/algorithms/mae.py]: done

**Review by fangyixiao18 — APPROVED** [review]

File: mmselfsup/models/heads/mae_head.py [hunk 2]
```diff
@@ -43,6 +43,22 @@ def patchify(self, imgs: torch.Tensor) -> torch.Tensor:
         x = x.reshape(shape=(imgs.shape[0], h * w, p**2 * 3))
         return x
 
+    def unpatchify(self, x: torch.Tensor) -> torch.Tensor:
+        """
```
Comment (tonysy) [inline mmselfsup/models/heads/mae_head.py]: Please add the comment for the purpose of this function

**Comment by tonysy** [comment]
The PR seems OK to me, pleas update the docstring as comments and fix the CI(test on windows).

**Review by tonysy — APPROVED** [review]

File: mmselfsup/models/heads/mae_head.py (hunk 2 above)
Comment (soonera) [reply mmselfsup/models/heads/mae_head.py]: Done

**Pull request merged**